src/plotting/plotting_utils.jl

```diff
@@ -0,0 +1,170 @@
+export plot_analysis_flow_parameters
+
+########################################################
+######## Flow-parameter analysing functions  ###########
+########################################################
+
+"""
+    vary_diff_plot(tspan, A_values, n_values, rgi_ids)
+
+Generate and plot the difference in ice thickness for a specified glacier over a time span `tspan`, for varying parameters `A_values` and `n_values`.
+
+# Arguments
+- `tspan::Tuple`: A tuple specifying the start and end years for the analysis.
+- `A_values::Array`: An array of A values (rate factor in Glen's flow law) to be used in the simulation.
+- `n_values::Array`: An array of n values (flow law exponent in Glen's flow law) to be used in the simulation.
+- `rgi_ids::Array`: An array containing the RGI (Randolph Glacier Inventory) ID of the glacier to be analyzed.
+
+# Returns
+- `Figure`: A Makie.jl figure object containing the generated plots.
+
+# Constraints
+- Supports a maximum grid size of 5x5 (lengths of `A_values` and `n_values` each should not exceed 5).
+- Only supports analysis for one glacier at a time (length of `rgi_ids` should be 1).
+"""
+
+function plot_analysis_flow_parameters(
+    tspan, 
+    A_values, 
+    n_values, 
+    rgi_ids, 
+    ice_thickness_source="Farinotti19", 
+    workers=1, 
+    use_iceflow=true, 
+    use_MB=true, 
+    use_multiprocessing=true, 
+    reltol=1e-8,
+    iceflow_model=SIA2Dmodel, 
+    mass_balance_model=TImodel1
+    )
+    # Calculate the size of the grid
```

Comment:
Could you please add identation here to clearly see where the code inside the function starts?

Reply:
Fixed indentation